salt/pillar/vault.py

```diff
@@ -79,16 +79,16 @@ def ext_pillar(minion_id,  # pylint: disable=W0613
         log.error('"%s" is not a valid Vault ext_pillar config', conf)
         return {}
 
+    vault_pillar = {}
+    
     try:
         path = paths[0].replace('path=', '')
         path = path.format(**{'minion': minion_id})
         url = 'v1/{0}'.format(path)
         response = __utils__['vault.make_request']('GET', url)
-        if response.status_code != 200:
-            response.raise_for_status()
-        vault_pillar = response.json()['data']
+        if response.status_code == 200:
+            vault_pillar = response.json()['data']
```

Comment:
Can you change this to a `.get()` just in case something in the future changes, and nothing is returned for some of these, I would rather have it do
```
vault_pillar = response.json().get('data', {})
```
Instead